runtime/doc/provider.txt

```diff
@@ -254,33 +254,25 @@ For Windows WSL, try this g:clipboard definition:
                 \   'cache_enabled': 0,
                 \ }
 <
-							    *clipboard-osc52*
+						   *clipboard-osc52* *g:osc52*
 Nvim bundles a clipboard provider that allows copying to the system clipboard
 using OSC 52. OSC 52 is an Operating System Command control sequence that
 writes the copied text to the terminal emulator. If the terminal emulator
 supports OSC 52 then it will write the copied text into the system clipboard.
 
-This is most useful when using Nvim remotely (e.g. via ssh) as Nvim does not
-have direct access to the system clipboard in that case.
-
-Because not all terminal emulators support OSC 52, this provider must be opted
-into explicitly by setting the following |g:clipboard| definition: >lua
-
-    vim.g.clipboard = {
-      name = 'OSC 52',
-      copy = {
-        ['+'] = require('vim.ui.clipboard.osc52').copy,
-        ['*'] = require('vim.ui.clipboard.osc52').copy,
-      },
-      paste = {
-        ['+'] = require('vim.ui.clipboard.osc52').paste,
-        ['*'] = require('vim.ui.clipboard.osc52').paste,
-      },
-    }
-<
+Nvim can attempt to automatically determine if the host terminal emulator
+supports the OSC 52 sequence and enable the OSC 52 clipboard provider if it
+does. This behavior is set by the |g:osc52| variable. If unset (nil) then Nvim
+will only attempt to detect OSC 52 support automatically when $SSH_TTY is set,
+$TMUX is unset, and 'clipboard' is not set to "unnamed" or "unnamedplus". If
+|g:osc52| is false, Nvim will never use the OSC 52 clipboard provider. If
+|g:osc52| is true, Nvim will always use the OSC 52 clipboard provider,
+regardless of the values of $SSH_TTY and $TMUX.
+
 Note that not all terminal emulators support reading from the system clipboard
 (and even for those that do, users should be aware of the security
-implications), so using OSC 52 for pasting may not be possible.
+implications), so using OSC 52 for pasting may not be possible. Users may need
```

Comment:
```suggestion
implications), so using OSC 52 for pasting may not be possible (and not necessary, because you can `|paste|` instead using your system paste function). Users may need
```